src/pip/_internal/operations/install/wheel.py

```diff
@@ -90,14 +90,13 @@ def fix_script(path: str) -> bool:
     assert os.path.isfile(path)
 
     with open(path, "rb") as script:
-        firstline = script.readline()
-        if not firstline.startswith(b"#!python"):
+        prelude = script.readline()
+        if (m := re.match(rb"^#!python[^\s]*(\s.*)?$", prelude)) is None:
```

Comment:
This doesn't conform to the spec, which *only* allows rewriting lines starting with precisely `b'#!python`.

Reply:
I think you misread: that’s exactly what that regex matches.

The only reason I switched from `startswith` to a regex is to capture the stuff behind the `#!python` and pass it on.

Reply:
Sorry, I did misread. But I'd still argue that the regex is harder to read. Why not simply split `firstline` on whitespace?

It's worth noting that there are some edge cases in the spec - `#!python3.12` should be rewritten, but it's not clear what "the correct interpreter" is if pip is running under Python 3.13 🙁

The current code doesn't get this right, either, but if we're going to claim the new code "is more robust" or "will no longer break", we should at least try to follow the spec.

Reply:
> Why not simply split firstline on whitespace?

and strip the trailing newline from the second part. I find the regex to be more readable than that.

Reply:
OK, we disagree.

(BTW, without documentation on `_get_shebang`, I don't know whether stripping the newline is necessary).